src/atc/delta/deltastream_handle.py

```diff
@@ -0,0 +1,218 @@
+            but skips any directories that begin with _. You can safely store
+            checkpoints alongside other data and metadata for a Delta table
+            using a directory structure such as <table_name>/_checkpoints
+            See: https://docs.databricks.com/structured-streaming/delta-lake.html
+
+        location: the location of the delta table (Optional)
+
+        data_format: the data format of the files that are read (Default delta)
+
+        """
+        super().__init__(name, location, data_format)
+
+        self._checkpoint_path = checkpoint_path
+        self._trigger_type = trigger_type.lower() if trigger_type else None
+        self._trigger_time = trigger_time.lower() if trigger_time else None
+        self._validate()
+        self._validate_trigger_type()
+        self._validate_checkpoint()
+
+    @classmethod
+    def from_tc(cls, id: str) -> "DeltaStreamHandle":
+        tc = Configurator()
+        return cls(
+            name=tc.table_property(id, "name", ""),
+            location=tc.table_property(id, "path", ""),
+            data_format=tc.table_property(id, "format", None),
+            checkpoint_path=tc.table_property(id, "checkpoint_path", None),
+        )
+
+    def _validate_trigger_type(self):
+        valid_trigger_types = {"availablenow", "once", "processingtime", "continuous"}
+        assert (
+            self._trigger_type in valid_trigger_types
+        ), f"Triggertype should either be {valid_trigger_types}"
+
+        # if trigger type is processingtime, then it should have a trigger time
+        assert (self._trigger_type == "processingtime") is (
+            self._trigger_time is not None
+        )
+
+    def _validate_checkpoint(self):
+        if "/_" not in self._checkpoint_path:
+            print(
+                "RECOMMENDATION: You can safely store checkpoints alongside "
+                "other data and metadata for a Delta table using a directory "
+                "structure such as <table_name>/_checkpoints"
+            )
+
+    def read(self) -> DataFrame:
+
+        reader = Spark.get().readStream.format("delta")
+
+        if self._location:
+            reader = reader.load(self._location)
+        else:
+            reader = reader.table(self._name)
+
+        return reader
+
+    def write_or_append(
+        self, df: DataFrame, mode: str, mergeSchema: bool = None
+    ) -> None:
+
+        assert mode in {"append", "overwrite", "complete"}
+        assert df.isStreaming
+
+        if mode == "overwrite":
+            print("WARNING: The term overwrite is called complete in streaming.")
+            mode = "complete"
+
+        writer = df.writeStream.option(
+            "checkpointLocation", self._checkpoint_path
+        ).outputMode(mode)
+
+        writer = self._add_trigger_type(writer)
+
+        writer = self._add_write_options(writer, mergeSchema)
+
+        writer.awaitTermination()  # Consider removing awaitTermination
```

Comment:
await is maybe necessary for the current ETL setup. 

We could make await optional? 

It is e.g. very difficult to test if await is not set.